File: frontend/web/src/pages/Intelligence/services/intelligence.service.ts
import { IntelligenceMapper } from "./intelligence.mapper";
import {
  MOCK_SUMMARY_DTO,
  MOCK_SELLERS_DTO,
  MOCK_RINGS_DTO,
  MOCK_PATTERNS_DTO,
  MOCK_IMAGE_CLUSTERS_DTO,
  MOCK_PHONE_CLUSTERS_DTO,
  MOCK_INVOICE_CLUSTERS_DTO,
  MOCK_MEMORY_INSIGHTS_DTO,
  MOCK_GRAPH_STATS_DTO
} from "./intelligence.mock";

const delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));

export const intelligenceService = {
  async getSummary() {
    await delay(600);
    return IntelligenceMapper.toSummary(MOCK_SUMMARY_DTO);
  },

  async getKnownSellers() {
    await delay(1000);
    return IntelligenceMapper.toSellers(MOCK_SELLERS_DTO);
  },

  async getFraudRings() {
    await delay(1200);
    return IntelligenceMapper.toRings(MOCK_RINGS_DTO);
  },

  async getKnownPatterns() {
    await delay(800);
    return IntelligenceMapper.toPatterns(MOCK_PATTERNS_DTO);
  },

  async getRepeatedImages() {
    await delay(1400);
    return IntelligenceMapper.toImageClusters(MOCK_IMAGE_CLUSTERS_DTO);
  },

  async getRepeatedPhones() {
    await delay(900);
    return IntelligenceMapper.toPhoneClusters(MOCK_PHONE_CLUSTERS_DTO);
  },

  async getRepeatedInvoices() {
    await delay(1100);
    return IntelligenceMapper.toInvoiceClusters(MOCK_INVOICE_CLUSTERS_DTO);
  },

  async getMemoryInsights() {
    await delay(1300);
    return IntelligenceMapper.toMemoryInsights(MOCK_MEMORY_INSIGHTS_DTO);
  },

  async getKnowledgeGraphStats() {
    await delay(700);
    return IntelligenceMapper.toGraphStats(MOCK_GRAPH_STATS_DTO);
  }
};

File: backend/services/intelligence_service.py
from typing import Dict, Any, List
from backend.database.repositories.intelligence_repo import IntelligenceRepository


class IntelligenceService:
    def __init__(self, repo: IntelligenceRepository):
        self._repo = repo

    def get_summary(self) -> Dict[str, Any]:
        return self._repo.get_summary()

    def get_sellers(self, limit: int = 50) -> List[Dict[str, Any]]:
        return self._repo.get_sellers(limit)

    def get_fraud_rings(self) -> List[Dict[str, Any]]:
        return self._repo.get_fraud_rings()

    def get_known_patterns(self) -> List[Dict[str, Any]]:
        return self._repo.get_known_patterns()

    def get_repeated_images(self) -> List[Dict[str, Any]]:
        return self._repo.get_repeated_images()

    def get_repeated_phones(self) -> List[Dict[str, Any]]:
        return self._repo.get_repeated_phones()

    def get_repeated_invoices(self) -> List[Dict[str, Any]]:
        return self._repo.get_repeated_invoices()

    def get_memory_insights(self) -> List[Dict[str, Any]]:
        return self._repo.get_memory_insights()
